### PR TITLE
fix-keyboard-test-case

### PR DESCRIPTION
**Pull request by ShakoHo**
modify the test case to able run it on aries... have dependency with bug 1205238
@zapion , @ypwalter r?

### CODE REVIEW

File: mtbf_driver/tests/mtbf/keyboard/test_mtbf_keyboard.py [hunk 1]
```diff
@@ -20,29 +20,27 @@ def test_keyboard_basic(self):
...
-        keyboard.choose_extended_character('A', 9)
+        # Now the menu would include the original char, so the index should +1
+        new_contact_form.keyboard.choose_extended_character('A', 9)
 
         # go back to app frame and finish this
         self.apps.switch_to_displayed_app()
         new_contact_form.tap_done()
+        self.wait_for_condition(lambda m: len(self.contacts_app.contacts) >= 1)
+
+        contact_details = self.contacts_app.contacts[0].tap()
```
Comment (zapion) [inline mtbf_driver/tests/mtbf/keyboard/test_mtbf_keyboard.py]: telephone number is randomly generated and contact list sorted by alphabet, supposedly index 0 will not match for most of the cases.

File: mtbf_driver/tests/mtbf/keyboard/test_mtbf_keyboard.py [hunk 2]
```diff
@@ -20,29 +20,27 @@ def test_keyboard_basic(self):
...
 
         # select special character using extended character selector
-        keyboard.choose_extended_character('A', 9)
+        # Now the menu would include the original char, so the index should +1
+        new_contact_form.keyboard.choose_extended_character('A', 9)
 
         # go back to app frame and finish this
         self.apps.switch_to_displayed_app()
         new_contact_form.tap_done()
+        self.wait_for_condition(lambda m: len(self.contacts_app.contacts) >= 1)
```
Comment (ypwalter) [inline mtbf_driver/tests/mtbf/keyboard/test_mtbf_keyboard.py]: suggest that you should get the quantity of contacts first and check if it +1

File: mtbf_driver/tests/mtbf/keyboard/test_mtbf_keyboard.py (hunk 1 above)
Comment (ypwalter) [inline mtbf_driver/tests/mtbf/keyboard/test_mtbf_keyboard.py]: how about try to check if contact of the same info exists and tap

File: mtbf_driver/tests/mtbf/keyboard/test_mtbf_keyboard.py (hunk 2 above)
Comment (ShakoHo) [inline mtbf_driver/tests/mtbf/keyboard/test_mtbf_keyboard.py]: good idea!

File: mtbf_driver/tests/mtbf/keyboard/test_mtbf_keyboard.py (hunk 1 above)
Comment (ShakoHo) [inline mtbf_driver/tests/mtbf/keyboard/test_mtbf_keyboard.py]: ok...I see...this line was came from gaiatest...they alway have one contact in their contact list...but we have more than 1... tap the info will have different behavior with gaiatest....maybe search the contact and then tap it..

**Comment by zapion** [comment]
r+

**Pull request merged**